src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -229,8 +384,42 @@ public DiskCache(string requestPath, string fullPath, string querystring)
+            if (!string.IsNullOrWhiteSpace(validatedVirtualCachePath))
+            {
+                // The cached file is valid so just rewrite the path.
+                context.RewritePath(this.virtualCachedFilePath, false);
+            }
+            else
+            {
+                // The file is outside of the web root so we cannot just rewrite the path since that won't work.
+                // We basically have a few options:
+
+                // 1. Create a custom VirtualPathProvider - this would work but we don't get all of the goodness that comes with
+                //      ASP.NET StaticFileHandler such as sending the correct cache headers, etc... you can see what I mean by 
+                //      looking at the source: https://referencesource.microsoft.com/#System.Web/StaticFileHandler.cs,492 
+                // 2. HttpResponse.TransmitFile - this actually uses the IIS/Windows kernel to do the transfer so it is very fast,
+                //      I've tested the header results and they are identical to the response headers set when using the StaticFileHandler
+                // 3. Set cache headers, etc... manually with regards to how the StaticFileHandler does it: 
+                        https://referencesource.microsoft.com/#System.Web/StaticFileHandler.cs,505
+                              // 4. Use reflection to invoke the StaticFileHandler somehow
+                              // 5. Use a custom StataicFileHandler like https://code.google.com/archive/p/talifun-web/wikis/StaticFileHandler.wiki 
+
+                // I've opted to go with the simplest solution and use TransmitFile, I've looked into the source of this and it uses the
+                // Windows Kernel, I'm not sure where in the source the headers get written but if you analyze the request/response headers when
+                // this is used, they are exactly the same as if the StaticFileHandler (i.e. RewritePath) is used, so this seems perfect and easy!                
+                context.Response.TransmitFile(CachedPath);
+
+                //This is quite important expecially if the request is a when an `IImageService` handles the request
+                // based on a custom request handler such as "database.axd/logo.png". If we don't end the request here
+                // and because we are not rewriting any paths, the request will continue to try to execute this handler
+                // and errors will occur. It should be fine that we are ending the request pipeline since
+                // all we really want to do here is send the file above. There's some arguments about using 
+                // `ApplicationInstance.CompleteRequest();` instead of Response.End() but in this case I believe it is 
+                // correct to use Response.End(), a good write-up of why this is can be found here:
+                // see: http://stackoverflow.com/a/36968241/694494
+                context.Response.End();
```

Comment:
Did you notice any performance implications or excess logging due to the `ThreadAbortException`?